src/Files.App/ViewModels/ShellViewModel.cs

```diff
@@ -1086,6 +1086,40 @@ await dispatcherQueue.EnqueueOrInvokeAsync(async () =>
 					}
 				}, cancellationToken);
 			}
+
+			return result is not null;
+		}
+
+		private async Task<bool> LoadThumbnailAsync(ListedItem item, IStorageItem matchingStorageItem, CancellationToken cancellationToken)
```

Comment:
Would it be worth implementing this logic as part of `FileThumbnailHelper`? This way the fix will apply throughout the app, and not just the file list.